src/pykeen/evaluation/evaluator.py

```diff
@@ -547,7 +553,8 @@ def evaluate(
 
     # Prepare for result filtering
     if filtering_necessary or positive_masks_required:
-        all_pos_triples = torch.cat([model.triples_factory.mapped_triples, mapped_triples], dim=0)
+        if all_pos_triples is None:
+            raise ValueError("When applying filtered evaluation, all_pos_triples needs to be provided.")
```

Comment:
We can assume that training triples are always known, right? I suggest to provide only the known triples beyond that, and bring back the `cat`.

Reply:
Done in scikit-learn.